### ISSUE TITLE
UP012: Description does not match website documentation, and some cases are missed

### ISSUE THREAD

**Issue by C0rn3j** [astral-sh/ruff#26854]
### Summary

```python
client.wfile.write((f"Fail: {e}\n").encode("utf-8"))

Unnecessary UTF-8 `encoding` argument to `encode`

help: Remove unnecessary `encoding` argumentRuff[UP012](https://docs.astral.sh/ruff/rules/unnecessary-encode-utf8)
```

In this example, in VSC I see the problem is the `encoding="utf-8"` argument, and I should just do `.encode()` instead.

The website however claims something else:

`Checks for unnecessary calls to encode as UTF-8.` - https://docs.astral.sh/ruff/rules/unnecessary-encode-utf8/

It wants me to remove the entire `encode()` function call and replace it with a literal `b"something"`.

Is the check then for:

A) Unnecessary use of encode() function  
B) Unnecessary default encoding argument  
C) Both  

?

If the answer is C, then the check is not working great, because this line produces no errors, whereas I think it should tell me to remove the encoding argument as it is default anyways:

```python
tb = str(exc).encode("utf-8").decode("unicode_escape").encode("latin1")
```

I target Python 3.11 if that matters.

### Version

ruff 0.15.21 (3e1f63645 2026-07-09)

**Comment by ntBre** [comment]
It looks like the check covers both A and B, and the diagnostic message was updated in https://github.com/astral-sh/ruff/pull/7393 to separate the two. The documentation still only covers A, though.

The reason that your second snippet isn't linted is because the rule is conservative about its string detection, and it doesn't recognize the result of the `str` call or the other chained operations as strings. The rule currently only applies to string and f-string literals. It seems like this was intentionally inherited from pyupgrade (https://github.com/asottile/pyupgrade/issues/138#issuecomment-496044965).

I think we should update the docs to reflect the two covered cases at least.

**Comment by nightcityblade** [comment]
Hi, I'd like to work on this. I'll submit a focused documentation fix shortly.
